src/result.rs

```diff
@@ -63,7 +79,13 @@ impl Printable for Error {
             self.kind,
         )?;
 
-        if let Some(self_val) = self.err.downcast_ref::<Error>() {
+        let any_ref = (*self.err).as_any();
+
+        type_to_trait!(&dyn AnyError, Printable);
```

Comment:
This is incorrect. `type_to_trait!` must be used only once for a type. i.e., as far as this PR is concerned, just in the test, for `PrintableErr`.